got error while compiling

root@server: cargo build --release
   Compiling tun2proxy v0.1.1 (/root/tun2proxy)
error[E0432]: unresolved import `std::os::fd::FromRawFd`
 --> src/setup.rs:7:5
  |
7 | use std::os::fd::FromRawFd;
  |     ^^^^^^^^^^^^^^^^^^^^^^ no `FromRawFd` in `os::fd`

error[E0603]: module `fd` is private
 --> src/setup.rs:7:14
  |
7 | use std::os::fd::FromRawFd;
  |              ^^ private module
  |
note: the module `fd` is defined here

error[E0599]: no function or associated item named `from_raw_fd` found for struct `File` in the current scope
   --> src/setup.rs:158:40
    |
158 |             let mut f = std::fs::File::from_raw_fd(fd);
    |                                        ^^^^^^^^^^^ function or associated item not found in `File`
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
1   | use std::os::unix::io::FromRawFd;
    |

warning: unused import: `Write`
 --> src/setup.rs:4:24
  |
4 | use std::io::{BufRead, Write};
  |                        ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

Some errors have detailed explanations: E0432, E0599, E0603.
For more information about an error, try `rustc --explain E0432`.
warning: `tun2proxy` (lib) generated 1 warning
error: could not compile `tun2proxy` due to 3 previous errors; 1 warning emitted

--- a/src/setup.rs
+++ b/src/setup.rs
@@ -8,7 +8,7 @@ use std::ffi::OsStr;
 use std::io::BufRead;
 use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 
-use std::os::fd::RawFd;
+use std::os::unix::io::RawFd;
 
 use std::process::{Command, Output};
 

